app/jobs/cache/browser_channels/responses_for_prefix.rb

```diff
@@ -0,0 +1,109 @@
+class Cache::BrowserChannels::ResponsesForPrefix
+  include Sidekiq::Worker
+  sidekiq_options queue: :low, retry: false
+
+  PATH = "channels/prefix/".freeze
+  PADDING_WORD = "P".freeze
+
+  attr_accessor :site_banner_lookups, :channel_responses, :temp_file
+
+  def perform(prefix)
+    generate_brotli_encoded_channel_response(prefix: prefix)
+    pad_file!
+    save_to_s3(prefix: prefix) unless Rails.env.test?
+    cleanup!
+  end
+
+  def generate_brotli_encoded_channel_response(prefix:)
+    @site_banner_lookups = SiteBannerLookup.where("sha2_base16 LIKE ?", prefix + "%")
+    begin
+      # Have to throw in a begin rescue block otherwise
+      # Zeitwerk::NameError (expected file $DIR/protos/channel_responses.rb to define constant ChannelResponses, but didn't)
+      # gets thrown.
+      require './protos/channel_responses'
+    rescue
+    end
+    @channel_responses = PublishersPb::ChannelResponseList.new
+    @site_banner_lookups.each do |site_banner_lookup|
+      channel_response = PublishersPb::ChannelResponse.new
+      channel_response.channel_identifier = site_banner_lookup.channel_identifier
+      channel_response.wallet_connected_state = site_banner_lookup.wallet_status
+      channel_response.wallet_address = site_banner_lookup.wallet_address if site_banner_lookup.wallet_address.present?
+      channel_response.site_banner_details = get_site_banner_details(site_banner_lookup)
+      channel_responses.channel_responses.push(channel_response)
+    end
+
+    json = PublishersPb::ChannelResponseList.encode(channel_responses)
+    info = Brotli.deflate(json)
+    @temp_file = Tempfile.new.binmode
+    # Write a 4-byte header saying the payload length
+    @temp_file.write([info.length].pack("L"))
+    @temp_file.write(info)
+    @temp_file.close
+    @temp_file
+  end
+
+  private
+
+  def cleanup!
+    begin
+      File.open(@temp_file.path, 'r') do |f|
+        File.delete(f)
+      end
+    rescue Errno::ENOENT
+    end
+  end
+
+  # We want to hide which file is being downloaded by making all requests be the same size
+  def pad_file!
+    path = @temp_file.path
+    # Round up to nearest KB
+    file_size = File.size(path)
+    # Converts size from like 326 to 1000
+    new_size = (((file_size + 1000) / 1000)) * 1000
```

Comment:
Nit: 1024 for KB?

Reply:
Let's keep it to base_10 for simplicity sake? It's not cut-and-dry: https://en.wikipedia.org/wiki/Kilobyte